video-recording.md

```diff
@@ -1 +1,62 @@
+# Record video/slideshow of running tests
 
+## General considerations
+
+* Limited to Firefox and Chrome, but supports both normal and headless mode. Taking plain images of these browsers in both modes already supported in the dedicated `chrome` and `firefox` browser providers by utilizing remote debugging protocols.
+* A stream of captured images transformed encoded into video format by [FFMPEG](https://www.ffmpeg.org/).
+* Should supports the following capturing modes:
```

Comment:
I propose to simplify the feature.

Let's remember the task: video recording is a way to understand the cause of failed tests.
So, it's a debug option. Because if customer turn on this feature for each test run it can strongly increase test execution time.

Patterns for path is unnecessary complication (If we will have `file pattern` feature before implementation then let it be).

In most cases:
- I want to see only failed tests. So, it's option by default.
- I don't want to rename the result filename. Let it be `video<timestamp>.mp4`
- I don't want to read  documentation about FFMPEG options. We should provide the video with minimal sufficient quality.

On test finished I can upload, rename and perform any actions with video.

After that we get the following examples of usage:
```js
testcafe chrome /tests --video-recording
testcafe chrome /tests --video-recording debug-video.mp4
testcafe chrome /tests --video-recording --video-recording-options mode=all-tests
testcafe chrome /tests --video-recording --video-recording-options mode=only-failed
```